tests/testthat/test-stackingutilities.R

```diff
@@ -37,3 +37,51 @@ test_that("ProcessAndStackDataForRegression", {
                                                  weights = weights),
                 "Size of variables doesn't agree")
 })
+
+
+
+# Test Stacking and Unstacking of Text and Categorization
+data.to.stack <- readRDS("text.analysis.stacking.rds")
+stacked.predicted.multi <- readRDS("stacked.predicted.categorization.multi.rds")
+stacked.predicted.multi <- stacked.predicted.multi$predicted
+stacked.predicted.single <- readRDS("stacked.predicted.categorization.single.rds")
+stacked.predicted.single <- stacked.predicted.single$predicted
+existings <- c("NULL", "binary.grid", "nominal.multi")
+texts <- c("text.multi", "multiple.text")
+weights <- subset <- rep(1, nrow(data.to.stack$text.multi))
+for (text.case in texts) {
+    for (existing.case in existings) {
+        existing <- if (existing.case == "NULL") NULL else data.to.stack[[existing.case]]
+        text <- data.to.stack[[text.case]]
+        test_that(paste0("StackTextAndCategorization: ", text.case, " : ", existing.case), {
+            expect_error(stacked <- StackTextAndCategorization(text = data.to.stack$text.multi, 
+                                    existing.categorization = existing,
+                                    weights = weights,
+                                    subset = subset),
+            NA)
+
+            # Correct dimensions
+            expect_equal(length(stacked$text), 1800)
+            expect_equal(length(stacked$weights), 1800)
+            expect_equal(length(stacked$subset), 1800)
+            expect_equal(length(stacked$inds), 1800)
+            if (existing.case == "binary.grid") {
+                expect_equal(nrow(stacked$existing), 1800)
+            } else if (existing.case == "nominal.multi") {
+                expect_equal(length(stacked$existing), 1800)
+            }
```

Comment:
The text currently only checks that it runs without error and that the lengths are correct. Not that the contents are stacked correctly. These tests could be improved to validate the internal elements? e.g. something like checking the stacked text output matches the unlisted original multi and the rownames are correctly constructed? and other similar checks for the weights and subset?